src/pages/courseDetailsPage/CourseDetailsPage.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { useEffect, useState } from "react";
+import { useLocation, useParams } from "react-router-dom";
+import { useGetCourseByIdQuery } from "redux/coursesApi";
+import {
+  ArrowIcon,
+  BackLink,
+  CourseDescription,
+  CourseLesson,
+  CourseLessonsList,
+  LessonTitle,
+  CourseTitle,
+  LockIcon,
+  Title,
+  TitleBox,
+  VideoContainer,
+  CurrentLesson,
+  CurrentLessContainer,
+  CurrentText,
+} from "./CourseDetailsPage.styled";
+import { Container } from "components/coursesList/CoursesList.styled";
+import VideoPlayer from "components/VideoPlayer/VideoPlayer";
+
+interface LessonseDetails {
+  id: string;
+  link: string;
+  title: string;
+  status: string;
+  previewImageLink: string;
+  order: number;
+}
+
+const CourseDetailsPage: React.FC = () => {
+  const { id: courseId = "" } = useParams();
+  const { state } = useLocation();
+  const backLinkHref = state?.from ?? "/";
+  const { data: course } = useGetCourseByIdQuery(courseId);
+  const [currentLesson, setCurrentLesson] = useState<LessonseDetails | null>(
+    null
+  );
+  const [isVideoOpen, setIsVideoOpen] = useState(true);
+
+  useEffect(() => {
+    if (course && course.lessons && course.lessons.length > 0) {
```

Comment:
```suggestion
    if (course?.lessons && course.lessons.length > 0) {
```
[link to article](https://learn.javascript.ru/optional-chaining#:~:text=%D0%9F%D1%80%D0%BE%D1%85%D0%BE%D0%B4%20%D0%BF%D1%80%D0%B8%20%D0%BF%D0%BE%D0%BC%D0%BE%D1%89%D0%B8%20%D0%BB%D0%BE%D0%B3%D0%B8%D1%87%D0%B5%D1%81%D0%BA%D0%BE%D0%B3%D0%BE%20%D0%BE%D0%BF%D0%B5%D1%80%D0%B0%D1%82%D0%BE%D1%80%D0%B0%20%D0%98%20%26%26%20%D1%87%D0%B5%D1%80%D0%B5%D0%B7%20%D0%B2%D0%B5%D1%81%D1%8C%20%D0%BF%D1%83%D1%82%D1%8C%20%D0%BA%20%D1%81%D0%B2%D0%BE%D0%B9%D1%81%D1%82%D0%B2%D1%83%20%D0%B3%D0%B0%D1%80%D0%B0%D0%BD%D1%82%D0%B8%D1%80%D1%83%D0%B5%D1%82%2C%20%D1%87%D1%82%D0%BE%20%D0%B2%D1%81%D0%B5%20%D0%BA%D0%BE%D0%BC%D0%BF%D0%BE%D0%BD%D0%B5%D0%BD%D1%82%D1%8B%20%D1%81%D1%83%D1%89%D0%B5%D1%81%D1%82%D0%B2%D1%83%D1%8E%D1%82%20(%D0%B5%D1%81%D0%BB%D0%B8%20%D0%BD%D0%B5%D1%82%2C%20%D0%B2%D1%8B%D1%87%D0%B8%D1%81%D0%BB%D0%B5%D0%BD%D0%B8%D0%B5%20%D0%BF%D1%80%D0%B5%D0%BA%D1%80%D0%B0%D1%89%D0%B0%D0%B5%D1%82%D1%81%D1%8F)%2C%20%D0%BD%D0%BE%20%D1%82%D0%B0%D0%BA%D0%B6%D0%B5%20%D0%BD%D0%B5%20%D1%8F%D0%B2%D0%BB%D1%8F%D0%B5%D1%82%D1%81%D1%8F%20%D0%B8%D0%B4%D0%B5%D0%B0%D0%BB%D1%8C%D0%BD%D1%8B%D0%BC.)

Reply:
Done